api/api.js

```diff
@@ -1480,7 +1482,42 @@ export default (router) => {
     }
   })
 
-  router.get('/hubs/:publicKeyOrHandle/posts', async (ctx) => {
+  const secondFloorMiddleware = async (ctx, next) => {
+    if (ctx.params.publicKeyOrHandle === 'FjAN2t3Q2URkTfCUupbbDoLPUzi5zCv8APDDj2XUcjoL') {
+      try {
+        return ratelimit({
+          driver: 'memory',
+          db: new Map(),
+          duration: 600000,
+          errorMessage:`Casey Jones you better watch your speed`,
+          id: (ctx) => {
+            return ctx.request.headers['x-id'] || '1'
+          },
+          headers: {
+            remaining: 'Rate-Limit-Remaining',
+            reset: 'Rate-Limit-Reset',
+            total: 'Rate-Limit-Total'
+          },
+          whitelist: (ctx) => {
+            if (
+              ctx.request.query.api_key === process.env.NINA_API_KEY
+            ) {
+              return true;
+            }
+        
+            return false;
+          },
+          max: 200,
+          disableHeader: false,
+        })(ctx, next)
+      } catch (error) {
+        console.log('secondFloorMiddleware error', error)
+      }
+    }
+    return next()
+  }
+
+  router.get('/hubs/:publicKeyOrHandle/posts', secondFloorMiddleware, async (ctx) => {
```

Comment:
just realized (if I'm reading this correctly) the issue with this is that we would run the secondFloor middleware every time a request to posts in hubs is made? or is that avoided by having the condition for FjAN2t3Q2URkTfCUupbbDoLPUzi5zCv8APDDj2XUcjoL here?

Reply:
the second floor middleware is run everytime, but it skips the ratelimiting if it matches FjAN2...

the other option would be to make an explicit route for that FjAN2 like you sketched out in discord but then we would need to copy the logic from the body in both the generic `/hubs/:publicKey/posts` and the `/hubs/FjAN.../posts` endpoint.  that could be a function but seems like the purpose of middleware is situations exactly like this